v2/cmd/controller/app/flags.go

```diff
@@ -8,83 +8,66 @@ package app
-	klog.InitFlags(flagSet)
-
-	var metricsAddr string
-	var profilingMetrics bool
-	var secureMetrics bool
-	var healthAddr string
-	var webhookPort int
-	var webhookCertDir string
-	var enableLeaderElection bool
-	var crdManagementMode string
-	var crdPatterns string
-	var preUpgradeCheck bool
+func InitFlags(flagSet *flag.FlagSet) Flags {
 
 	// default here for 'MetricsAddr' is set to "0", which sets metrics to be disabled if 'metrics-addr' flag is omitted.
-	flagSet.StringVar(&metricsAddr, "metrics-addr", "0", "The address the metric endpoint binds to.")
-	flagSet.BoolVar(&secureMetrics, "secure-metrics", true, "Enable secure metrics. This secures the pprof and metrics endpoints via Kubernetes RBAC and HTTPS")
-	flagSet.BoolVar(&profilingMetrics, "profiling-metrics", false, "Enable pprof metrics, only enabled in conjunction with secure-metrics. This will enable serving pprof metrics endpoints")
+	metricsAddr := flagSet.String("metrics-addr", "0", "The address the metric endpoint binds to.")
+	secureMetrics := flagSet.Bool("secure-metrics", true, "Enable secure metrics. This secures the pprof and metrics endpoints via Kubernetes RBAC and HTTPS")
+	profilingMetrics := flagSet.Bool("profiling-metrics", false, "Enable pprof metrics, only enabled in conjunction with secure-metrics. This will enable serving pprof metrics endpoints")
+
+	healthAddr := flagSet.String("health-addr", "", "The address the healthz endpoint binds to.")
+	webhookPort := flagSet.Int("webhook-port", 9443, "The port the webhook endpoint binds to.")
+	webhookCertDir := flagSet.String("webhook-cert-dir", "", "The directory the webhook server's certs are stored.")
 
-	flagSet.StringVar(&healthAddr, "health-addr", "", "The address the healthz endpoint binds to.")
-	flagSet.IntVar(&webhookPort, "webhook-port", 9443, "The port the webhook endpoint binds to.")
-	flagSet.StringVar(&webhookCertDir, "webhook-cert-dir", "", "The directory the webhook server's certs are stored.")
-	flagSet.BoolVar(&enableLeaderElection, "enable-leader-election", false,
-		"Enable leader election for controllers manager. Enabling this will ensure there is only one active controllers manager.")
-	flagSet.StringVar(&crdManagementMode, "crd-management", "auto",
+	enableLeaderElection := flagSet.Bool("enable-leader-election", false, "Enable leader election for controllers manager. Enabling this will ensure there is only one active controllers manager.")
+
+	crdManagementMode := flagSet.String("crd-management", "auto",
 		"Instructs the operator on how it should manage the Custom Resource Definitions. One of 'auto', 'none'")
-	flagSet.StringVar(&crdPatterns, "crd-pattern", "", "Install these CRDs. CRDs already in the cluster will also always be upgraded.")
-	flagSet.BoolVar(&preUpgradeCheck, "pre-upgrade-check", false,
+	crdPatterns := flagSet.String("crd-pattern", "", "Install these CRDs. CRDs already in the cluster will also always be upgraded.")
+	preUpgradeCheck := flagSet.Bool("pre-upgrade-check", false,
```

Comment:
I'm expecting a conflic there once #4217 merges, but that should be easy to resolve.

Reply:
Resolved.